numpy/_array_api/_set_functions.py

```diff
@@ -0,0 +1,14 @@
+from __future__ import annotations
+
+from ._types import Tuple, Union, array
+from ._array_object import ndarray
+
+import numpy as np
+
+def unique(x: array, /, *, return_counts: bool = False, return_index: bool = False, return_inverse: bool = False) -> Union[array, Tuple[array, ...]]:
```

Comment:
You'd want to avoid returning unsafe unions as much as possible, as they're quite frankly a pain to deal with (case in point: https://github.com/numpy/numpy/issues/18305).
This seems like a prime case for an overload:

```suggestion
@overload
def unique(x: array, /, *, return_counts: Literal[False] = ..., return_index: Literal[False] = ..., return_inverse: Literal[False] = ...) -> array: 
    ...
@overload
def unique(x: array, /, *, return_counts: bool = ..., return_index: bool = ..., return_inverse: bool = ...) -> Tuple[array, ...]: 
    ...
def unique(x, /, *, return_counts=False, return_indexFalse, return_inverse=False):
```

Reply:
`unique` itself is likely to be modified in the spec so that it doesn't have a polymorphic return type. See https://github.com/data-apis/array-api/issues/212. So I am going to leave this as it is for now, and update it once we decide on the better variant in the spec.